File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
import java.util.*;

class Solution {
    public static int[] solution(String s) {
        List<Integer> answer = new ArrayList<>();
        List<Character> arr = new ArrayList<>();

        for (char c : s.toCharArray()) {
            if (arr.contains(c)) {
                for (int j = arr.size() - 1; j >= 0; j--) {
                    if (c == arr.get(j)) {
                        answer.add(arr.size() - j);
                        break;
                    }
                }
            } else {
                answer.add(-1);
            }
            arr.add(c);
        }

        // List를 int[] 배열로 변환
        int[] result = new int[answer.size()];
        for (int i = 0; i < answer.size(); i++) {
            result[i] = answer.get(i);
        }

        return result;
    }
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.kt
class Solution {
    fun solution(s: String): IntArray {
        var answer: IntArray = intArrayOf()
        var arr = mutableListOf<Char>()

        for (i in s) {
            if (arr.contains(i)) {
                for (j in arr.size - 1 downTo 0) {

                    if (i == arr[j]) {
                        answer += arr.size - j
                        break
                    }
                }

            } else {
                answer += -1
            }
            arr.add(i)
        }

        return answer
    }
}